library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/generator/GafferPopEdgeGeneratorTest.java

```diff
@@ -50,13 +50,13 @@ public void shouldConvertGafferEdgeToGafferPopReadOnlyEdge() {
         final GafferPopEdge gafferPopEdge = generator._apply(edge);
 
         // Then
-        assertEquals(TestGroups.EDGE, gafferPopEdge.label());
-        assertEquals(source, gafferPopEdge.outVertex().id());
-        assertEquals(dest, gafferPopEdge.inVertex().id());
-        assertEquals(propValue, gafferPopEdge.property(TestPropertyNames.STRING).value());
-        assertEquals(1, Lists.newArrayList(gafferPopEdge.properties()).size());
-        assertSame(graph, gafferPopEdge.graph());
-        assertTrue(gafferPopEdge.isReadOnly());
+        assertThat(gafferPopEdge.label()).isEqualTo(TestGroups.EDGE);
+        assertThat(gafferPopEdge.outVertex().id()).isEqualTo(source);
+        assertThat(gafferPopEdge.inVertex().id()).isEqualTo(dest);
+        assertThat(gafferPopEdge.property(TestPropertyNames.STRING).value()).isEqualTo(propValue);
+        assertThat(Lists.newArrayList(gafferPopEdge.properties()).size()).isEqualTo(1);
```

Comment:
This should probably be:
```suggestion
        assertThat(gafferPopEdge.properties()).hasSize(1);
```
It seems like a mistake that this was ever wrapped in a list. Doing this means the size will always be one as the list will have one item, that item being the actual size.

Reply:
Ah makes sense, I figured that how I implemented it wasn't quite right but wasn't sure on the best fix. So that makes sense :) thanks

Reply:
What I put above is actually misleading. I misread the brackets, `gafferPopEdge.properties()` is an `Iterator` so it does need to be converted in order to determine the size. Ideally we'd not use `Lists` for this but use Assertj instead.

Reply:
I will add new comments with suggestions.